File: Application/src/main/java/com/dfm/Users/Services/IUsersService.java
package com.dfm.Users.Services;

import com.dfm.Users.Entities.UserEntity;
import com.dfm.Users.Entities.UserRoleEntity;
import com.dfm.Users.Entities.Dto.UserResponseDto;
import com.dfm.Users.Entities.Dto.UserRoleResponseDto;

public interface IUsersService
{
    public UserResponseDto authenticate(UserEntity user_entity);
    public UserRoleResponseDto readByUserRole(UserRoleEntity user_role_entity);
    public UserResponseDto register(UserEntity user_entity);
}

File: FrontEnd/Projects/Blazor/Users/Services/IUsersService.cs
using Blazor.Users.Entities;
using Blazor.Users.Entities.Dto;

namespace Blazor.Users.Services;

public interface IUsersService
{
    public UserEntity user_entity { get; }
    public string token { get; }

    public Task initialize();
    public Task authenticate(UserEntity user_entity);
    public Task<UserRoleResponseDto> readByUserRoleAsync(UserRoleEntity user_role_entity);
    public Task register(UserEntity user_entity);
    public Task logout();
}
